pkg/storage/imagestore/imagestore.go

```diff
@@ -0,0 +1,2177 @@
+						Str("delay", imgStore.gcDelay.String()).Msg("gc: failed to check if blob is older than delay")
+
+					return err
+				}
+
+				if canGC {
+					imgStore.log.Info().Str("repository", repo).Str("digest", desc.Digest.String()).
+						Msg("gc: removing manifest without tag")
+
+					if err := imgStore.deleteImageManifest(repo, desc.Digest.String(), true); err != nil {
+						if errors.Is(err, zerr.ErrManifestConflict) {
+							imgStore.log.Info().Str("repository", repo).Str("digest", desc.Digest.String()).
+								Msg("gc: skipping removing manifest due to conflict")
+
+							continue
+						}
+
+						return err
+					}
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func gcCosignReferences(imgStore *ImageStore, index *ispec.Index, repo string,
+	cosignDescriptors []ispec.Descriptor,
+) error {
+	for _, cosignDesc := range cosignDescriptors {
+		foundSubject := false
+		// check if we can find the manifest which the reference points to
+		for _, desc := range index.Manifests {
+			// signature
+			subject := fmt.Sprintf("sha256-%s.%s", desc.Digest.Encoded(), remote.SignatureTagSuffix)
+			if subject == cosignDesc.Annotations[ispec.AnnotationRefName] {
+				foundSubject = true
+			}
+
+			// sbom
+			subject = fmt.Sprintf("sha256-%s.%s", desc.Digest.Encoded(), remote.SBOMTagSuffix)
+			if subject == cosignDesc.Annotations[ispec.AnnotationRefName] {
+				foundSubject = true
+			}
+		}
+
+		if !foundSubject {
+			// remove manifest
+			imgStore.log.Info().Str("repository", repo).Str("digest", cosignDesc.Digest.String()).
+				Msg("gc: removing cosign reference without subject")
+
+				// no need to check for manifest conflict, if one doesn't have a subject, then none with same digest will have
+			if err := imgStore.deleteImageManifest(repo, cosignDesc.Digest.String(), false); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func gcOCIArtifacts(imgStore *ImageStore, index *ispec.Index, repo string,
+	artifactManifests []extendedManifest,
+) error {
+	for _, artifactManifest := range artifactManifests {
+		foundSubject := false
+
+		for _, desc := range index.Manifests {
+			if desc.Digest == artifactManifest.Subject.Digest {
+				foundSubject = true
+			}
+		}
+
+		if !foundSubject {
+			// remove manifest
+			imgStore.log.Info().Str("repository", repo).Str("digest", artifactManifest.Digest.String()).
+				Str("artifactType", zcommon.GetManifestArtifactType(artifactManifest.Manifest)).
+				Msg("gc: removing oci artifact without subject")
+
+			// no need to check for manifest conflict, if one doesn't have a subject, then none with same digest will have
+			if err := imgStore.deleteImageManifest(repo, artifactManifest.Digest.String(), false); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func isBlobOlderThan(imgStore storageTypes.ImageStore, repo string,
+	digest godigest.Digest, delay time.Duration, log zerolog.Logger,
```

Comment:
Why are we using zerolog.Logger instead of our own implementation of logger? It's used in multiple places in the storage package..